src/braket/pennylane_plugin/ahs_device.py

```diff
@@ -329,15 +323,32 @@ def _run_task(self, ahs_program):
         return task
 
     def _validate_pulses(self, pulses):
-
-        # ToDo: allow local drive
-
-        if len(pulses) > 1:
-            raise NotImplementedError(
-                f"Multiple pulses in a Rydberg Hamiltonian are not currently supported on "
-                f"hardware. Recieved {len(pulses)} pulses.")
-
-        if pulses[0].wires != self.wires:
-            raise NotImplementedError(
-                f"Only global drive is currently supported on hardware. Found drive defined for subset "
-                f"{[pulses[0].wires]} of all wires [{self.wires}]")
\ No newline at end of file
+        # Iterate through pulses to find global drive
+        max_wires = 0
+        for i, pulse in enumerate(pulses):
+            if len(pulse.wires) > max_wires:
+                max_wires = len(pulse.wires)
+                global_index = i
+
+        local_pulses = pulses.copy()
+        global_pulse = local_pulses.pop(global_index)
```

Comment:
Maybe this function should actually return these two things, since we will now need them inside the `apply` function to decide which parts get converted into a DrivingField vs a ShiftingField

Reply:
In that case, would it make sense to also process the pulses in `_validate_pulses` such that getting the pattern is easier?